src/Controller/FrontendUsersProfileController.php

```diff
@@ -111,23 +115,62 @@ private function getUserRecord(ContentType $contentType): Content
     {
         /** @var User $user */
         $user = $this->getUser();
+        
+        // Access user record, if available
+        $contentTypeSlug = $this->getExtension()->getExtConfig('contenttype', $user->getRoles()[0]);
+        $contentType = $this->getBoltConfig()->getContentType($contentTypeSlug);
+
         $content = $this->contentRepository->findBy([
             'author' => $user,
             'contentType' => $contentType->getSlug(),
         ]);
-
+        
+        // If user record unavailable, create it
         if (empty($content)) {
-            $content = new Content($contentType);
-            $content->setAuthor($user);
-            $content->setPublishedAt(new \DateTime());
-            $content->setStatus(Statuses::PUBLISHED);
-            $contentTypeName = strtolower($contentType->get('name', $contentType->get('slug')));
-            $content->setContentType($contentTypeName);
-            $this->contentFillListener->fillContent($content);
-        } elseif (is_iterable($content)) {
+            return $this->new($contentType);
+        } 
+        elseif (is_iterable($content)) {
             $content = end($content);
         }
+        
+        return $content;
+    }
 
+    // Define the user record content
+    private function new(ContentType $contentType): Content
+    {
+        /** @var User $user */
+        $user = $this->getUser();
+        $contentTypeSlug = $this->getExtension()->getExtConfig('contenttype', $user->getRoles()[0]);
+        $contentType = $this->getBoltConfig()->getContentType($contentTypeSlug);
+        
+        $content = new Content($contentType);
+        $content->setAuthor($user);
+        $content->setCreatedAt(new \DateTime());
+        $content->setPublishedAt(new \DateTime());
+        $content->setStatus(Statuses::PUBLISHED);
+        $contentTypeName = strtolower($contentType->get('name', $contentType->get('slug')));
+        $content->setContentType($contentTypeName);
+        $content->setFieldValue('displayName', $user->getDisplayName()); // Hidden field for record title
+        $content->setFieldValue('username', $user->getUsername()); // Hidden field with copy of username
+        
+        // Initialise ALL extra fields as defined in the contenttype with empty strings.
+        // This ensures they are displayed on the /profile/edit route without backend intervention
+        $content->setFieldValue('dob', '');
```

Comment:
Do we need this `dob`?

Reply:
@I-Valchev we don't need it. I just matched it to the ContentType demonstration in the docs to show how to initialise any custom extra fields. Perhaps we can comment it out?

Reply:
Ah, gotcha. Maybe we should comment it out and in future initialize all the fields in the specified ContentType, or none. I'll comment it out for now.

Reply:
Roger that, @I-Valchev!